### PR TITLE
AsciiDoc reporter clean-ups

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details. 

> **Note**
> This is a **breaking change** as the `XHtmlTemplateReporter` reporter is removed in favor of the almost identical `HtmlTemplateReporter` reporter.

### CODE REVIEW

**Review by fviernau — APPROVED** [review]

File: README.md
Comment (nnobelis) [inline README.md:1]: I think the argument of XHML "being more modern" is fallacious. Could you justify it ?
My understanding is that, _currently_, XHTML is 'just' HTML with XML syntax.
But the current state is pretty complicated, e.g. https://softwareengineering.stackexchange.com/questions/149839/is-xhtml5-dead-or-is-it-just-an-synonym-of-html5.

Comment (sschuberth) [reply README.md:1]: Note that when saying "HTML" here, I don't mean "HTML5". I'm referring to the "original" HTML (version <= 4.01) to which XHTML IMO is an improvement as it's more strict. Actually, HTML5 is probably more close to XHTML than to HTML <= 4.01.

Anyway, while I personally linked the syntax in `xhtml-template-reporter-expected-result.xhtml` a little bit better than the one in `html-template-reporter-expected-result.html`, I don't really care that much. I can also remove `XHtmlTemplateReporter` instead, if that's the consensus.

Comment (nnobelis) [reply README.md:1]: Both are fine for me. My comment what just about the justification of the changes.

Comment (sschuberth) [reply README.md:1]: > I can also remove `XHtmlTemplateReporter` instead, if that's the consensus.

After reading through https://docs.asciidoctor.org/asciidoctor/latest/html-backend/, I guess that's what I'll do. We could later make it a `HtmlTemplateReporter`-specific option to output XHTML instead, if required, via the `htmlsyntax=xml` attribute.

Comment (nnobelis) [reply README.md:1]: Maybe adjust the commit message with "XHTML is stricter" or "XHTML has a better syntax" ?
In the second case you make is clear it's a personal preference.

**Comment by nnobelis** [comment]
@sschuberth This is not a breaking change anymore, right ?

**Comment by sschuberth** [comment]
> @sschuberth This is not a breaking change anymore, right ?

It is, for any previous user of the `XHtmlTemplateReporter`.

**Review by fviernau — APPROVED** [review]

**Pull request merged**